tracing/span.go

```diff
@@ -443,8 +457,109 @@ type Headers struct {
+func (h Headers) AnySet() bool {
+	return h.TraceID != "" ||
+		h.SpanID != "" ||
+		h.Flags != "" ||
+		h.Sampled != nil
+}
+
+// TraceIDUint attempts to convert h.TraceID into a uint64, if it succeeds it
+// returns the value and 'true'.  If it fails, either because h.TraceID is not
+// set or it is malformed, ok will be 'false' and you should not rely on the ID
+// returned.
+//
+// If h.TraceID was malformed, an error will be logged using the global tracer's
+// logger but no error will be returned.
+func (h Headers) TraceIDUint() (id uint64, ok bool) {
+	if h.TraceID == "" {
+		return
+	}
+
+	var err error
+	id, err = strconv.ParseUint(h.TraceID, 10, 64)
+	if err != nil {
+		globalTracer.getLogger()(fmt.Sprintf(
+			"Malformed trace id in http ctx: %q, %v",
+			h.TraceID,
+			err,
+		))
+		return
+	}
+	ok = true
+	return
+}
+
+// SpanIDUint attempts to convert h.SpanID into a uint64, if it succeeds it
+// returns the value and 'true'.  If it fails, either because h.SpanID is not
+// set or it is malformed, ok will be 'false' and you should not rely on the ID
+// returned.
+//
+// If h.SpanID was malformed, an error will be logged using the global tracer's
+// logger but no error will be returned.
+func (h Headers) SpanIDUint() (id uint64, ok bool) {
+	if h.SpanID == "" {
+		return
+	}
+
+	var err error
+	id, err = strconv.ParseUint(h.SpanID, 10, 64)
+	if err != nil {
+		globalTracer.getLogger()(fmt.Sprintf(
+			"Malformed span id in http ctx: %q, %v",
+			h.SpanID,
+			err,
+		))
+		return
+	}
+	ok = true
+	return
+}
+
+// FlagsInt attempts to convert h.Flags into an int64, if it succeeds it
+// returns the value and 'true'.  If it fails, either because h.Flags is not
+// set or it is malformed, ok will be 'false' and you should not rely on the ID
+// returned.
+//
+// If h.Flags was malformed, an error will be logged using the global tracer's
+// logger but no error will be returned.
+func (h Headers) FlagsInt() (flags int64, ok bool) {
+	if h.Flags == "" {
+		return
+	}
+
+	var err error
+	flags, err = strconv.ParseInt(h.Flags, 10, 64)
+	if err != nil {
+		globalTracer.getLogger()(fmt.Sprintf(
+			"Malformed flags in http ctx: %q, %v",
+			h.Flags,
+			err,
+		))
+		return
+	}
+	ok = true
+	return
+}
+
+// SampledBool returns the boolean value of h.Sampled and a flag specifying
+// whether h.Sampled was set or not.  If it not set, both "sampled" and "ok"
+// will return "false" but that does not mean that "sampled" should be false, you
+// should only used the returned value for "sampled" if "ok" is true.
+func (h Headers) SampledBool() (sampled bool, ok bool) {
+	if h.Sampled == nil {
+		return false, false
+	}
+	return *h.Sampled, true
+}
+
 // StartSpanFromHeaders creates a server span from the passed in Headers.
 //
+// This should be used when you are initializing a server span *from headers*.
+// If you do not have headers to initialize your span from, you should use
+// StartTopLevelServerSpan instead.
```

Comment:
nit: Since we already handled that in this function (the added first `if`), this should be reworded?